api-report/purchases-js.api.md

```diff
@@ -176,10 +176,34 @@ export interface Price {
 // @public
 export interface Product {
     readonly currentPrice: Price;
+    readonly defaultSubscriptionOption: SubscriptionPurchaseOption | null;
     readonly displayName: string;
     readonly identifier: string;
     readonly normalPeriodDuration: string | null;
     readonly presentedOfferingIdentifier: string;
+    readonly subscriptionPurchaseOptions: {
+        [optionId: string]: SubscriptionPurchaseOption;
+    };
+}
+
+// @public
+export interface PurchaseFlowParams {
+    customerEmail?: string;
+    htmlTarget?: HTMLElement;
+    purchaseOption?: PurchaseOption | null;

```

Comment:
I made some extra changes to the API here. So instead of passing an ID, now we pass an entire `PurchaseOption` object. I believe this is preferable to passing strings around.

Reply:
does it need to be nullable and optional? Or only optional works?

Reply:
I made it both for convenience... Basically if you make it only optional but not nullable, you can't make it receive a null value like:
`purchase({ purchaseOption: possiblyNullablePurchaseOption })`... Honestly, I'm not super familiar with TS conventions either, but I thought this was more flexible.